cc3d/core/PyCoreSpecs.py

```diff
@@ -5309,6 +5379,114 @@ def from_xml(cls, _xml: CC3DXMLElement):
         return cls()
 
 
+class SurfaceTrackerPlugin(_PyCorePluginSpecs, _PyCoreSteerableInterface):
+    """ SurfaceTracker Plugin """
+
+    name = "surface_tracker"
+    registered_name = "SurfaceTracker"
+
+    def __init__(self,
+                 neighbor_order: int = None,
+                 max_neighbor_order: int = None,
+                 max_neighbor_distance: float = None):
+
+        super().__init__()
+
+        self.max_neighbor_distance = max_neighbor_distance
+        self.max_neighbor_order = max_neighbor_order
+        self.neighbor_order = neighbor_order
+
+    @property
+    def xml(self) -> ElementCC3D:
+        """
+        CC3DXML element; generated from specification dictionary
+
+        :return: CC3DML XML element
+        """
+        self._el = self.generate_header()
+        if self.neighbor_order is not None:
+            self._el.ElementCC3D('NeighborOrder', {}, str(self.neighbor_order))
+        elif self.max_neighbor_order is not None:
+            self._el.ElementCC3D('MaxNeighborOrder', {}, str(self.max_neighbor_order))
+        elif self.max_neighbor_distance is not None:
+            self._el.ElementCC3D('MaxNeighborDistance', {}, str(self.max_neighbor_distance))
+        return self._el
+
+    @classmethod
+    def from_xml(cls, _xml: CC3DXMLElement):
+        """
+        Instantiate an instance from a CC3DXMLElement parent instance
+
+        :param _xml: parent xml
+        :return: python class instace
+        :rtype: MomentOfInertiaPlugin
+        """
+        el = cls.find_xml_by_attr(_xml)
+
+        neighbor_order = None
+        max_neighbor_order = None
+        max_neighbor_distance = None
```

Comment:
What are the default values if these are not specified in the XML?

Reply:
`None` values here correspond to no specification in the XML. Each has their own effect, but what's defined here should be consistent with analogous XML spec.